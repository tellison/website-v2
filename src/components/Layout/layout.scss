// Here you can customize bootstrap
$primary: #14003c;
$secondary: #dadada;

@import "~bootstrap/scss/bootstrap";

/* set a max-width for horizontal fluid layout and make it centered */
.container {
  margin-right: auto;
  margin-left: auto;
  max-width: 60rem;
}

body {
  --black4: #cbd4d9;
  --black9: #2c3437;
  --brand4: #c52158;
  --brand1: #14003c;
  --page-pink: #fcf4f6;
  --bg: white;
  --text-normal: #222;
  --color-fill-banner: #9992;
  --color-text-primary: var(--black9);
  --color-text-accent: var(--color-brand-primary);

  background-color: var(--bg);

  .support-matrix th {
    background-color: #aeadad97;
  }
}

.text-pink {
  color: var(--brand4);
}

.halign-center {
  text-align: center;
}

.valign-middle {
  vertical-align: middle;
}

.bg-pink {
  background-color: var(--brand4) !important;
}

.bg-purple {
  background-color: var(--brand1) !important;
}

.bg-grey {
  background-color: rgb(237, 234, 234) !important;
}

.table-secondary {
  --bs-table-bg: #333;
}

body:not(.dark) .dark-mode-only,
.dark .light-mode-only {
  display: none;
}

.hide-on-mobile {
  @media only screen and (max-width: 500px) {
    display: none !important;
  }
}

.adopt-demo-container {
  position: relative;
}

.adopt-demo {
  background: #1f272a;
  border-radius: 0.4rem;
  box-shadow: 5px 10px 50px rgba(0, 0, 0, 25%);
  color: #fff;
  height: 450px;
  max-width: 832px;
  width: 90vw;
  z-index: 2;
}

.callout {
  padding: 20px;
  margin: 20px 0;
  border: 1px solid #eee;
  border-left-width: 5px;
  border-radius: 3px;

  h4 {
    margin-top: 0;
    margin-bottom: 5px;
  }

  p:last-child {
    margin-bottom: 0;
  }

  code {
    border-radius: 3px;
  }

  & + .bs-callout {
    margin-top: -5px;
  }
}

@each $name,$color in (default,#777),(primary,#428bca),
  (success,#5cb85c),(danger,#d9534f),(warning,#f0ad4e),
  (info,#5bc0de), (bdc,#29527a) {
  .callout-#{$name} {
    border-left-color: $color;

    h4 {
      color: $color;
    }
  }
}

.w-20 {
  width: 20% !important;
}

.table-borderless td,
.table-borderless th {
  border: 0;
  padding: 0.1em 0.5rem;

  .table td,
  .table th,
  .table a {
    font-size: 0.9rem;
  }
}

.cc-color-override-1434350510.cc-window {
  background-color: var(--brand4) !important;
}

iframe {
  width: 100%;
  height: 25em;
}
