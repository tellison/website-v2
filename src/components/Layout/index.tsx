import * as React from 'react'
import PropTypes from 'prop-types'
import { useStaticQuery, graphql } from 'gatsby'
import SSRProvider from 'react-bootstrap/SSRProvider'

import NavBar from '../NavBar'
import Banner from '../Banner'
import Footer from '../Footer'
import './layout.scss'

const Layout = ({ children }) => {
  const data = useStaticQuery(graphql`
    query SiteTitleQuery {
      site {
        siteMetadata {
          title
        }
      }
    }
  `)

  return (
    <SSRProvider>
      <div
        className='container-fluid p-0'
        style={{
          backgroundColor: 'var(--bg)',
          color: 'var(--text-normal)',
          transition: 'color 0.2s ease-out, background 0.2s ease-out'
        }}
      >
        <NavBar />
        <Banner />
        <main>
          {children}
        </main>
        <Footer />
      </div>
    </SSRProvider>
  )
}

Layout.propTypes = {
  children: PropTypes.node.isRequired
}

export default Layout
