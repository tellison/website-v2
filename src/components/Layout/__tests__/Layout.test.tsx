import React from 'react';
import { render } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest'
import Layout from '..';

// Mock the Banner component as this changes frequently
vi.mock('../../Banner', () => {
  return {
    default: () => <div>mock-banner</div>,
  };
})

// Mock the NavBar component as this is tested separately
vi.mock('../../NavBar', () => {
  return {
    default: () => <div>mock-navbar</div>,
  };
})

// Mock the Footer component as this is tested separately
vi.mock('../../Footer', () => {
  return {
    default: () => <div>mock-footer</div>,
  };
})

describe('Layout component', () => {
  it('renders correctly with data', () => {
    const { container } = render(
      <Layout>
        mock-children
      </Layout>
    );
    expect(container).toMatchSnapshot();
  });
});