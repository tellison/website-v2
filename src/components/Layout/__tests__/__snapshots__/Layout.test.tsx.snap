// Vitest Snapshot v1

exports[`Layout component > renders correctly with data 1`] = `
<div>
  <div
    class="container-fluid p-0"
    style="transition: color 0.2s ease-out, background 0.2s ease-out;"
  >
    <div>
      mock-navbar
    </div>
    <div>
      mock-banner
    </div>
    <main>
      mock-children
    </main>
    <div>
      mock-footer
    </div>
  </div>
</div>
`;
